app/scripts/modules/core/src/managed/Environments.tsx

```diff
@@ -22,20 +23,35 @@ interface IEnvironmentsProps {
   app: Application;
 }
 
-export function Environments(props: IEnvironmentsProps) {
-  const { app } = props;
+export function Environments({ app }: IEnvironmentsProps) {
   const dataSource: ApplicationDataSource<IManagedApplicationEnvironmentSummary> = app.getDataSource('environments');
-  const { data: environments } = useDataSource(dataSource);
+  const {
+    data: { environments, artifacts, resources },
+  } = useDataSource(dataSource);
+
   const [selectedArtifact, setSelectedArtifact] = useState<ISelectedArtifact>();
   const [isFiltersOpen] = useState(false);
 
-  const totalContentWidth = isFiltersOpen ? CONTENT_WIDTH + 248 + 'px' : CONTENT_WIDTH + 'px';
+  const resourcesById = useMemo(() => keyBy(resources, 'id'), [resources]);
+  const resourcesByEnvironment = useMemo(
+    () =>
+      environments.reduce((byEnvironment, { name, resources: resourceIds }) => {
+        byEnvironment[name] = resourceIds.map(id => resourcesById[id]);
+        return byEnvironment;
+      }, {} as { [environment: string]: IManagedResourceSummary[] }),
+    [environments, resourcesById],
```

Comment:
i'd like to:
1) in the immediate term, push this calculation down to `ArtifactDetail` so it's done less frequently/only when necessary
2) in the medium term, move this mostly to the backend by storing a list of resources inside these artifact version summaries